src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -171,8 +171,12 @@ public void actionPerformed(ActionEvent e) {
             UserPanel.this, "Enter user name:", "Add User", JOptionPane.PLAIN_MESSAGE,
             null, null, "");
         if (s != null && s.length() > 0) {
-          clientContext.user.addUser(s);
-          UserPanel.this.getAllUsers(listModel);
+          if(clientContext.user.addUser(s)==true){
```

Comment:
General whitespace nit:
according to the style guide, if statements should look like this:
if (foo) {
  ...
} else {
  ...
}